crates/rust-client/src/sync/state_sync.rs

```diff
@@ -458,6 +458,8 @@ pub async fn on_note_received(
 ) -> Result<bool, ClientError> {
     let note_id = *committed_note.note_id();
 
+    let tags = store.get_unique_note_tags().await?;
```

Comment:
We shouldn't use all tags for this, only tags that were added manually by the user. 
```suggestion
    let tags = store.get_note_tags()?.into_iter()..map(|r| r.source == NoteTagSource::User)
```